src/parquet/file/metadata.h

```diff
@@ -178,6 +178,8 @@ class PARQUET_EXPORT FileMetaData {
   // Return const-pointer to make it clear that this object is not to be copied
   const SchemaDescriptor* schema() const;
 
+  std::unordered_map<std::string, std::string> key_value_metadata() const;
```

Comment:
One reason to keep `unordered_map` is so that we don't have to redefine another version of this class in `arrow-cpp`.